### PR TITLE
Handle Tessa service failures

### LINKED ISSUE #16: Handle failures to talk to Tessa more elegantly
At the moment a failure to talk to Tessa will result in the page not even loading.

### PR DESCRIPTION
**Pull request by jpowell**
closes watermarkchurch/tessa#16

> - asset has been uploaded, but service is unavailable - we'd probably want to display to the user that something is there, but can't be retrieved
> - asset has not been uploaded and service is unavailable - fail silently, render as if service is up
> - asset exists in client DB, but not in Tessa service DB - act as if no asset exists and allow to be overwritten?

### CODE REVIEW

File: spec/tessa/model_spec.rb [hunk 1]
```diff
@@ -327,6 +327,81 @@
...
+
+        context "argument is array" do
+          let(:arg) { [ id, id * 2 ] }
+
+          it "returns array" do
+            expect(result).to be_a(Array)
+          end
+
+          it "returns instances of FailedAsset" do
+            expect(result).to all( be_a(Tessa::FailedAsset) )
```
Comment (jpowell) [inline spec/tessa/model_spec.rb]: @tpett you seen the `all` collection expectation?

File: lib/tessa/failed_asset.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module Tessa
+  class FailedAsset
+    include Virtus.model
+
+    attribute :id, Integer
+    attribute :error, StandardError
```
Comment (jpowell) [inline lib/tessa/failed_asset.rb]: had to use `StandardError` because of load order. Thoughts?

File: spec/tessa/model_spec.rb (hunk 1 above)
Comment (tpett) [inline spec/tessa/model_spec.rb]: I think I've seen it in your code before. Pretty nice.

**Comment by jpowell** [comment]
in testing scenarios locally, it seems as though the service is throwing a 500 when it should be throwing a 400. a little more inspection forthcoming.....

**Pull request merged**